src/rgw/rgw_dedup_cluster.cc

```diff
@@ -1147,24 +1249,35 @@ namespace rgw::dedup {
                                   dedup_req_type_t dedup_type,
                                   const DoutPrefixProvider *dpp)
   {
-    librados::IoCtx ioctx;
-    int ret = init_dedup_pool_ioctx(store->getRados(), dpp, ioctx);
-    if (ret != 0) {
-      return ret;
-    }
+    ldpp_dout(dpp, 1) << __func__ << "::dedup_type = " << dedup_type << dendl;
 
     dedup_epoch_t old_epoch;
     // store the previous epoch for cmp-swap
-    ret = get_epoch(ioctx, dpp, &old_epoch, __func__);
+    int ret = get_epoch(store, dpp, &old_epoch, __func__);
     if (ret != 0) {
-      return ret;
+      // generate an empty epoch with zero counters
```

Comment:
this is a change in behavior.
what could be the reason for the failure? do we always create an empty epoch regardless of why we failed?